openpgp/packet/private_key.go

```diff
@@ -426,6 +426,8 @@ func (pk *PrivateKey) parsePrivateKey(data []byte) (err error) {
 		return pk.parseECDHPrivateKey(data)
 	case PubKeyAlgoEdDSA:
 		return pk.parseEdDSAPrivateKey(data)
+	case PubKeyAlgoBadElGamal:
+		return errors.UnsupportedError("parsing el-gamal sign-or-encrypt privatekeys is unsupported")
```

Comment:
maybe just return unsupported as a `default:` case, even if not `el-gamal`?

Reply:
Done. I don't think you can get here without going through a function in public_key.go that validates its one of these, but agree that this is more defensive since go isn't checking enum handling.